internal/llm/agent/setup-agent.go

```diff
@@ -0,0 +1,57 @@
+package agent
+
+import (
+	"context"
+	"fmt"
+	"github.com/sst/opencode/internal/config"
+	"github.com/sst/opencode/internal/llm/models"
+	"github.com/sst/opencode/internal/llm/provider"
+	"github.com/sst/opencode/internal/message"
+)
+
+type setupAgent struct {
+}
+
+func NewSetupAgent() (Service, error) {
```

Comment:
The UI heavily relies on the `PrimaryAgent` being present, this agent is just to stub all functionality until the setup is complete.